desk/lib/bolt.hoon

```diff
@@ -162,72 +160,56 @@
     =^  cards  yosh  (on-fail:ag term tang)
     [cards this]
   --
+  ::
   |%
-  ++  permissions
-    |_  [=bowl:gall state=state-0]
-    +*  parent  (sein:title our.bowl now.bowl src.bowl)
+  ++  is-allowed
+    |=  =bowl:gall
+    ^-  ?
+    ::
+    =*  parent  (sein:title our.bowl now.bowl src.bowl)
+    =*  user    src.bowl
+    =*  moon-gud  |(=(our.bowl parent) (~(has in whitelist) parent))
+    =*  moon-bad  |(=(our.bowl parent) (~(has in blacklist) parent))
+    ::
+    ::  Do not allow if in blacklist or kids and kid of blacklister
+    ?&  |(!(~(has in blacklist) user) &(kids !moon-bad))
+        ::  If whitelist is off -- allow user,
+        ?|  !white
+        ::  If whitelist is on, check if user is in whitelist or kid of whitelister.
+            =(our.bowl user)
+            (~(has in whitelist) user)
+            &(kids moon-gud)
+    ==  ==
   ::
-    ++  is-allowed
-      |=  user=@p
-      =+  state
-      =*  moon-gud  |(=(our.bowl parent) (~(has in users.whitelist) parent))
-      =*  moon-bad  |(=(our.bowl parent) (~(has in users.blacklist) parent))
-      ^-  ?
-      ::
-      ::  Do not allow if in blacklist or kids and kid of blacklister
-      ?&  |(!(~(has in users.blacklist) user) &(kids !moon-bad))
-      ::
-      :: If whitelist is off -- allow user,
-          ?:  =(which %|)
-            %&
-      ::
-      :: If whitelist is on, check if user is in whitelist or kid of whitelister.
-          ?|  =(our.bowl user)
-              (~(has in users.whitelist) user)
-              &(kids moon-gud)
-          ==
-      ==
-      ::
-    --
   ++  json
     =,  enjs:format
-    |_  state=state-0
-  ::
-    ++  website
+    |%
+    ++  emit
       |=  [type=cord diff=^json]     
       ^-  card
       %-  fact:agentio  :_  ~[/website]
       :-  %json  !>  ^-  ^json
-    ::
-    ::  parsing starts here
+      ::
+      ::  parsing starts here
       %-  pairs
       :~  [%type s+type]
           [%diff diff]
-        :-  %state
-        %-  pairs
-        :~  [%which b+which]
-            [%kids b+kids]
-          :-  %whitelist
-          %-  pairs
```

Comment:
If you only want to encode one pair, use `+frond:enjs:format` instead of `+pairs:enjs:format`.